### PR TITLE
Added Section to "Gold - Combinatorics"

### PR DESCRIPTION
**Pull request by dutinmeow**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: content/4_Gold/Combinatorics.mdx [hunk 1]
```diff
@@ -289,6 +289,103 @@ int main() {
...
+{n \choose k}(n-k)! = \frac{n!}{k!(n-k)!}(n-k)! = \frac{n!}{k!}
+$$
+
+Thus, the problem now becomes computing 
+
+$$
+n!\sum_{k=0}^n\frac{(-1)^k}{k!}
+$$
+
+which can be done in linear time
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:320]:
```suggestion
which can be done in linear time.
```

File: content/4_Gold/Combinatorics.mdx [hunk 2]
```diff
@@ -289,6 +289,103 @@ int main() {
...
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+#include <atcoder/modint>
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:327]: what

File: content/4_Gold/Combinatorics.mdx [hunk 3]
```diff
@@ -289,6 +289,103 @@ int main() {
...
+int main() {
+	using namespace std;
+
+	int N, M;
+	cin >> N >> M;
+	mint::set_mod(M);
+	mint c = 1;
+	for (int i = 1; i <= N; i++) 
+		cout << (c = (c * i) + (i % 2 == 1 ? -1 : 1)).val() << ' ';
+	cout << '\n';
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:340]:
```suggestion
	cout << endl;
```

File: content/4_Gold/Combinatorics.mdx [hunk 4]
```diff
@@ -289,6 +289,103 @@ int main() {
...
+
+int main() {
+	using namespace std;
+
+	int N, M;
+	cin >> N >> M;
+	mint::set_mod(M);
+	mint c = 1;
+	for (int i = 1; i <= N; i++) 
+		cout << (c = (c * i) + (i % 2 == 1 ? -1 : 1)).val() << ' ';
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:339]: braces

File: content/4_Gold/Combinatorics.mdx [hunk 5]
```diff
@@ -289,6 +289,103 @@ int main() {
...
+which can be computed in linear time with a simple DP. The base cases are that $!0 = 1$ and $!1 = 0$. 
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+#include <atcoder/modint>
+
+using mint = atcoder::modint;
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:368]: same, what

File: content/4_Gold/Combinatorics.mdx [hunk 6]
```diff
@@ -289,6 +289,103 @@ int main() {
...
+	cin >> N >> M;
+	mint::set_mod(M);
+	mint a = 1, b = 0;
+	cout << 0 << ' ';
+	for (int i = 2; i <= N; i++) {
+		mint c = (i - 1) * (a + b);
+		cout << c.val() << ' ';
+		a = b, b = c;
+	}
+	cout << '\n';
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:383]:
```suggestion
	cout << endl;
```

File: content/4_Gold/Combinatorics.mdx [hunk 7]
```diff
@@ -289,6 +289,103 @@ int main() {
...
+
+	int N, M;
+	cin >> N >> M;
+	mint::set_mod(M);
+	mint a = 1, b = 0;
+	cout << 0 << ' ';
+	for (int i = 2; i <= N; i++) {
+		mint c = (i - 1) * (a + b);
+		cout << c.val() << ' ';
+		a = b, b = c;
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:381]: separate lines

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/4_Gold/Combinatorics.mdx (hunk 5 above)
Comment (dutinmeow) [reply content/4_Gold/Combinatorics.mdx:370]: Yosupo has atcoder library builtin

Comment (SansPapyrus683) [reply content/4_Gold/Combinatorics.mdx:370]: then at least add a comment saying "hey, this is something builtin to the library"
also add a link to the sourcecode

Comment (dongliu0426) [reply content/4_Gold/Combinatorics.mdx:370]: https://github.com/atcoder/ac-library/blob/master/atcoder/modint.hpp

Comment (dongliu0426) [reply content/4_Gold/Combinatorics.mdx:370]: link this?

**Review by dongliu0426 — APPROVED** [review]

File: content/4_Gold/Combinatorics.mdx (hunk 5 above)
Comment (dutinmeow) [reply content/4_Gold/Combinatorics.mdx]: ok, linked

File: content/4_Gold/Combinatorics.mdx [hunk 8]
```diff
@@ -324,8 +324,9 @@ which can be done in linear time.
 
 ```cpp
 #include <bits/stdc++.h>
+		
+// https://atcoder.github.io/ac-library/document_en/modint.html
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:328]: say that this is included in the grader

File: content/4_Gold/Combinatorics.mdx [hunk 9]
```diff
@@ -289,6 +289,108 @@ int main() {
...
+
+```cpp
+#include <bits/stdc++.h>
+		
+// https://atcoder.github.io/ac-library/document_en/modint.html (included in grader)
+#include <atcoder/modint>
+using mint = atcoder::modint;
+
+int main() {
+	using namespace std;
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:333]: why on god's green earth is using namespace here

Comment (dutinmeow) [reply content/4_Gold/Combinatorics.mdx:333]: if there's no function ambiguity in the main function, then it's fine to include namespace std

Comment (SansPapyrus683) [reply content/4_Gold/Combinatorics.mdx:333]: nevermind that this is the only place in the entire codebase where it's here

Comment (dutinmeow) [reply content/4_Gold/Combinatorics.mdx:333]: if this matters for you then change it

**Pull request merged**